plugins/secretstores/gdchauth/gdchauth.go

```diff
@@ -0,0 +1,276 @@
+package gdchauth
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/x509"
+	_ "embed"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+/* #nosec G101 */
+const (
+	tokenExchangeType       = "urn:ietf:params:oauth:token-type:token-exchange"
+	accessTokenTokenType    = "urn:ietf:params:oauth:token-type:access_token"
+	serviceAccountTokenType = "urn:k8s:params:oauth:token-type:serviceaccount"
+)
```

Comment:
Please directly use those constants in the code instead of declaring them here! This puts everything in one place and also gets rid of the linter issue.